core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -551,9 +553,26 @@ private JarBuildItem buildThinJar(CurateOutcomeBuildItem curateOutcomeBuildItem,
                 jars.add(path);
             }
         }
+
+        /*
+         * There are some files like META-INF/microprofile-config.properties that usually don't exist in application
+         * and yet are always looked up (spec compliance...) and due to the location in the jar,
+         * the RunnerClassLoader needs to look into every jar to determine whether they exist or not.
+         * In keeping true to the original design of the RunnerClassLoader which indexes the directory structure,
+         * we just add a fail-fast path for files we know don't exist.
+         *
+         * TODO: if this gets more complex, we'll probably want a build item to carry this information instead of hard
+         * coding it here
```

Comment:
Yeah, that was my question. I also wonder if we should index the files in `META-INF/` (I'm not sure if we should index recursively or not) and if we don't do it recursively `META-INF/services/`?

That would cover this use case too.

Reply:
I have another [draft PR](https://github.com/quarkusio/quarkus/issues/13256) exploring services, so I would prefer not to do anything with services for the time being.

The thing is, that even if we do index all files in `META-INF`, https://github.com/quarkusio/quarkus/blob/6542a037e3ed1457cf7b211ca5038930d3c828fa/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java#L123 still is an obstacle. 
Which is why I think the what I did here is the simplest thing to do for now, as we don't yet know how far we want to take this.

Reply:
> Yeah, that was my question. I also wonder if we should index the files in `META-INF/` (I'm not sure if we should index recursively or not) and if we don't do it recursively `META-INF/services/`?

There are HTML/JS/CSS files in `META-INF/resources` which probably won't need to be indexed

Reply:
Indeed. They didn't show up in the profiling I did, but in the future we certainly should look further